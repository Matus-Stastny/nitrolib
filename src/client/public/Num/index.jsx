/* @flow */
import withIntl from "../services/intl/withIntl";

type Props = {
  value: number,
  locale: string,
  format?: { [key: string]: number | string },
};

const Num = (props: Props) => new Intl.NumberFormat(props.locale, props.format).format(props.value);

Num.defaultProps = {
  format: {},
};

export default withIntl(Num);
