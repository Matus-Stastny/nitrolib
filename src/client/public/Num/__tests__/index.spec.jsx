/* @flow */
import * as React from "react";
import { shallow } from "enzyme";

import Num from "../index";

describe("#Num", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<Num value={1234} />);

    expect(wrapper).toMatchSnapshot();
  });
});
