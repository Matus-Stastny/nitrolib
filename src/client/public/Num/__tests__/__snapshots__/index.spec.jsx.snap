// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Num should render correctly 1`] = `
<Num
  format={Object {}}
  locale="en"
  translate={[Function]}
  value={1234}
/>
`;
